util/graph.js

```diff
@@ -0,0 +1,77 @@
+/** Class that signifies a graph */
+class Graph {
+    /**
+      * The underlying graph object. This is an object with nodes and edges.
+      * The class contains an adjacency list property that represents the edges
+     * between each of the nodes. Each edge can have attributes such as weight (which 
+     * by default is 10) and whether or not the edge is directed.
+     * 
+      */
+    adjacencyList;
+ 
+    constructor() {
+       this.adjacencyList = []
```

Comment:
https://github.com/tair/mb4-service/pull/58/files#r1170649826
"Using an array is not suboptimal - because in order to get the edge for two nodes, you have to traverse the second level.

Prefer Maps instead. This way the graph can be properly represented by an adjacency list:
Map<node, Map<node, edge>>"